.github/workflows/test.yml

```diff
@@ -47,7 +47,50 @@ jobs:
         make docker-test-integration
     - name: Upload cAdvisor log file
       uses: actions/upload-artifact@v1
-      if: failure()
+      if: success() || failure()
       with:
-        name: cadvisor.log
-        path: ${{ github.workspace }}/go/src/github.com/google/cadvisor/cadvisor.log
+        name: cadvisor-${{ matrix.distro }}-${{ matrix.arch }}-${{ matrix.go-versions}}-${{ github.run_number }}-${{ github.run_id }}.log
+        path: ${{ github.workspace }}/cadvisor.log
+  test-integration-non-amd64:
+    strategy:
+      matrix:
+        go-versions: [ 1.19, 1.18 ]
+        arch: [ aarch64, armv7 ]
+        distro: [ ubuntu20.04 ]
```

Comment:
Could you use the `platform` variable like in test, and test-integration jobs?
Or even stay with distro but let's have it the same for all jobs :)